tests/app/celery/test_scheduled_tasks.py

```diff
@@ -624,3 +631,27 @@ def test_trigger_link_does_nothing_if_cbc_proxy_disabled(
         trigger_link_tests()
 
     assert mock_trigger_link_test.called is False
+
+
+@freeze_time('2021-07-19 15:50')
+@pytest.mark.parametrize('status, finishes_at, final_status', [
+    (BroadcastStatusType.BROADCASTING, '2021-07-19 16:00', BroadcastStatusType.BROADCASTING),
+    (BroadcastStatusType.BROADCASTING, '2021-07-19 15:40', BroadcastStatusType.COMPLETED),
+    (BroadcastStatusType.BROADCASTING, None, BroadcastStatusType.BROADCASTING),
+    (BroadcastStatusType.PENDING_APPROVAL, None, BroadcastStatusType.PENDING_APPROVAL),
+    (BroadcastStatusType.CANCELLED, '2021-07-19 15:40', BroadcastStatusType.CANCELLED),
+])
+def test_auto_expire_broadcast_messages(
+    status,
+    finishes_at,
+    final_status,
+    sample_template
+):
+    message = create_broadcast_message(
+        status=status,
+        finishes_at=finishes_at,
+        template=sample_template,
+    )
+
+    auto_expire_broadcast_messages()
+    assert message.status == final_status
```

Comment:
Could consider adding an assertion for the `updated_at` time here since that can also change

Reply:
I did think about it, but it would make the parametrization more complex.

Fortunately it's no longer in scope as we discussed in https://github.com/alphagov/notifications-api/pull/3294#discussion_r673913382.